Jenkinsfile

```diff
@@ -4,49 +4,45 @@
  * "Sonatype" is a trademark of Sonatype, Inc.
  */
 @Library(['private-pipeline-library', 'jenkins-shared']) _
-import com.sonatype.jenkins.pipeline.GitHub
-import com.sonatype.jenkins.pipeline.OsTools
 
 properties([
   parameters([
-    booleanParam(defaultValue: false, description: 'Force Red Hat Certified Build for a non-main branch', name: 'force_red_hat_build'),
-    booleanParam(defaultValue: false, description: 'Skip Red Hat Certified Build', name: 'skip_red_hat_build'),
-    string(defaultValue: '', description: 'Override automatic version assignment', name: 'version')
-  ])
+    string(
+      name: 'version',
+      description: 'Version tag to apply to the image, like 1.142.0-1.'
+    ),
+  ]),
 ])
 
 node('ubuntu-zion') {
-  def version, isMain
-  def organization = 'sonatype',
-      archiveName = 'nxiq-operator-certified-metadata.zip'
-
-  stage('Preparation') {
-    deleteDir()
-
-    def checkoutDetails = checkout scm
-
-    isMain = checkoutDetails.GIT_BRANCH in ['origin/main', 'main']
-
-    version = params.version ?: readVersion()
-  }
-
-  stage('Trigger Red Hat Certified Image Build') {
-    if ((! params.skip_red_hat_build) && (isMain || params.force_red_hat_build)) {
+  try {
+    stage('Preparation') {
+      deleteDir()
+
+      checkout scm
+
+      sh 'docker system prune -a -f'
+      sh '''
+        wget -q -O preflight \
+          https://github.com/redhat-openshift-ecosystem/openshift-preflight/releases/download/1.4.0/preflight-linux-amd64
+        chmod 755 preflight
+      '''
+    }
+    stage('Build') {
       withCredentials([
-          string(credentialsId: 'operator-nxiq-rh-build-project-id', variable: 'PROJECT_ID'),
-          string(credentialsId: 'rh-build-service-api-key', variable: 'API_KEY')]) {
-        runGroovy('ci/TriggerRedHatBuild.groovy', "'${version}' '${PROJECT_ID}' '${API_KEY}'")
+        string(
+          credentialsId: 'red-hat-scan-registry-password-nexus-iq-operator',
+          variable: 'REGISTRY_PASSWORD'),
+        string(
+          credentialsId: 'red-hat-api-token',
+          variable: 'API_TOKEN'),
+      ]) {
+        sh 'PATH="$PATH:." VERSION=$version ./build_red_hat_image.sh'
```

Comment:
The same as in the other PR, don't you need to pass the `REGISTRY_PASSWORD` too as a parameter?

Reply:
as mentioned in the other PR, `withCredentials` is importing it as REGISTRY_PASSWORD environment variable, which matches the name the script wants. (little `version` environment variable that the parameter added didn't match big `VERSION`, so I needed to explicitly set it here.)